pkg/common/x509svid/csr.go

```diff
@@ -30,5 +30,12 @@ func ValidateCSR(csr *x509.CertificateRequest, validationMode idutil.ValidationM
 		return errors.New("CSR must have exactly one URI SAN")
 	}
 
-	return idutil.ValidateSpiffeIDURL(csr.URIs[0], validationMode)
+	id, err := spiffeid.FromURI(csr.URIs[0])
+	if err != nil {
+		return fmt.Errorf("CSR SPIFFE ID %q is invalid: %w", csr.URIs[0], err)
```

Comment:
Maybe CSR _with_ SPIFFE ID is more accurate and reads better?
Same with the next error.
```suggestion
		return fmt.Errorf("CSR with SPIFFE ID %q is invalid: %w", csr.URIs[0], err)
```

Reply:
sgtm!